backend/dataall/modules/s3_datasets_shares/services/dataset_sharing_service.py

```diff
@@ -126,11 +126,11 @@ def verify_dataset_share_objects(uri: str, share_uris: list):
         with get_context().db_engine.scoped_session() as session:
             for share_uri in share_uris:
                 share = ShareObjectRepository.get_share_by_uri(session, share_uri)
```

Comment:
list_shareable_items is an ugly complex query that is not needed here. Instead I replaced it by `get_all_share_items_in_share`